pandas/core/indexes/datetimes.py

```diff
@@ -1991,15 +1992,17 @@ def tz_localize(self, tz, ambiguous='raise', errors='raise'):
         DatetimeIndex(['2018-03-01', '2018-03-02', '2018-03-03'],
         dtype='datetime64[ns]', freq='D')
 
-        #localize DatetimeIndex in US?Eastern time zone.
+        localize DatetimeIndex in US/Eastern time zone.
```

Comment:
The explanations of the examples need improvements.  The first paragraph is outdated (end date is not the 8th). And I don't think you need to say that, it's clear from the code. So I would just get rid of it.

In this case, it should start with a capital `L`. May be you can add here what you were explaining before, that `dti` is naive (and also what you're already saying, that you're localizing to US/Eastern). Personally I'd rename `dti` to `tz_aware`, so it's more obvious.

Reply:
You are saying this : 

In [2]: tz_aware = pd.date_range('2018-03-01', '2018-03-03')

In [3]: tz_aware
Out[3]: DatetimeIndex(['2018-03-01', '2018-03-02', '2018-03-03'], dtype='datetime64[ns]', freq='D')

In [4]: tz_aware.tz_localize(tz='US/Eastern')
Out[4]: 
DatetimeIndex(['2018-03-01 00:00:00-05:00', '2018-03-02 00:00:00-05:00',
               '2018-03-03 00:00:00-05:00'],
              dtype='datetime64[ns, US/Eastern]', freq='D')

In [5]: tz_aware.tz_localize(None)
Out[5]: DatetimeIndex(['2018-03-01', '2018-03-02', '2018-03-03'], dtype='datetime64[ns]', freq='D')

Reply:
Almost. But in `In [2]` that's not a `tz_aware` index, it's a naive one. And in `In [4]` you do want to save the output, as that's the aware one that you'll one to convert to naive in `In [5]`.

Make sure that the example makes sense for you too. Reviewers can also be wrong.

Reply:
but thats why we added comment before where we analyse aware and naive

Reply:
I agree, but I think the variable names can still help by being descriptive, even if you explain what you're doing. Feel free to keep `dti` if you prefer it over `tz_naive`. But for me it's more clear to see what the example does, if the variables are more descriptive. All objects in the example could be `dti`, and it doesn't help much IMO, but naming one `tz_naive` and the other `tz_aware`, makes it clear what every object used has.

Sorry if I'm not being able to explain what I mean clearer enough.

Reply:
I hope this commit define exactly what you want in example section

Reply:
sorry for disappointing you :(